### ISSUE TITLE
mustache-2.4.0 failed to build in Stackage Nightly

### ISSUE THREAD

**Issue by juhp** [JustusAdam/mustache#58]
```
/var/stackage/work/unpack-dir/unpacked/mustache-2.4.0-44c4d43ecfe1fee11fb03ffd49b0580ed00eec5144067092801ef4256df77ef8/src/Text/Mustache/Internal/Types.hs:395:1: error:
    • Overlapping instances for Lift TemplateCache
        arising from a use of ‘lift’
      Matching instances:
        two instances involving out-of-scope types
        (use -fprint-potential-instances to see them all)
    • In the second argument of ‘Language.Haskell.TH.Lib.Internal.appE’, namely
        ‘(lift x2_aclX)’
      In the expression:
        (Language.Haskell.TH.Lib.Internal.appE
           ((Language.Haskell.TH.Lib.Internal.appE
               ((Language.Haskell.TH.Lib.Internal.appE
                   (Language.Haskell.TH.Lib.Internal.conE
                      ((Name (mkOccName "Template"))
                         (((NameG DataName)
                             (mkPkgName "mustache-2.4.0-7gu0iJzW3shKWbIZjtf98g"))
                            (mkModName "Text.Mustache.Internal.Types")))))
                  (lift x0_aclV)))
              (lift x1_aclW)))
          (lift x2_aclX)
      In a case alternative:
          Template x0_aclV x1_aclW x2_aclX
            -> (Language.Haskell.TH.Lib.Internal.appE
                  ((Language.Haskell.TH.Lib.Internal.appE
                      ((Language.Haskell.TH.Lib.Internal.appE
                          (Language.Haskell.TH.Lib.Internal.conE
                             ((Name (mkOccName "Template"))
                                (((NameG DataName)
                                    (mkPkgName "mustache-2.4.0-7gu0iJzW3shKWbIZjtf98g"))
                                   (mkModName "Text.Mustache.Internal.Types")))))
                         (lift x0_aclV)))
                     (lift x1_aclW)))
                 (lift x2_aclX)
    |
395 | deriveLift ''Template
    | ^^^^^^^^^^^^^^^^^^^^^

/var/stackage/work/unpack-dir/unpacked/mustache-2.4.0-44c4d43ecfe1fee11fb03ffd49b0580ed00eec5144067092801ef4256df77ef8/src/Text/Mustache/Internal/Types.hs:397:10: error:
    • Overlapping instances for Lift TemplateCache
        arising from a use of ‘Language.Haskell.TH.Syntax.$dmlift’
      Matching instances:
        two instances involving out-of-scope types
        (use -fprint-potential-instances to see them all)
    • In the expression:
        Language.Haskell.TH.Syntax.$dmlift
          @('ghc-prim-0.7.0:GHC.Types.LiftedRep) @(TemplateCache)
      In an equation for ‘lift’:
          lift
            = Language.Haskell.TH.Syntax.$dmlift
                @('ghc-prim-0.7.0:GHC.Types.LiftedRep) @(TemplateCache)
      In the instance declaration for ‘Lift TemplateCache’
    |
397 | instance Lift TemplateCache where
    |          ^^^^^^^^^^^^^^^^^^
```

**Comment by Bodigrim** [comment]
This is caused by new instances in `unordered-containers-0.2.17.0`, https://github.com/haskell-unordered-containers/unordered-containers/pull/343

**Comment by Bodigrim** [comment]
As a Hackage trustee, I made necessary revisions:
https://hackage.haskell.org/package/mustache-2.3.1/revisions/
https://hackage.haskell.org/package/mustache-2.3.2/revisions/
https://hackage.haskell.org/package/mustache-2.4.0/revisions/

**Comment by hasufell** [comment]
The resolver seems to prefer mustache-2.3.0 now, so it can pick unordered-containers-2.7. However, that leads to build failures and users need to specify bounds explicitly: https://github.com/commercialhaskell/stack/issues/5298#issuecomment-835378436

Note that cabal has two options here:

1. pick mustache later than 2.3.0 and unordered-containers earlier than 2.7
2. pick mustache lower than 2.3.1 and unordered-containers 2.7

**Comment by JustusAdam** [comment]
Presumably it should be possible to fix this by adding a `VERSION` guard to the `deriveLift` line. I will see if I can find the time to look into that but if anyone wants to take a stab it should be easy to add , just takes a bit to test it works.

**Comment by JustusAdam** [comment]
This was fixed in #59.

I will prepare a release to publish the fix

**Comment by juhp** [comment]
Thanks!
